sdks/python/apache_beam/typehints/typehints.py

```diff
@@ -1190,6 +1190,12 @@ def normalize(x, none_as_type=False):
     return type(None)
   elif x in _KNOWN_PRIMITIVE_TYPES:
     return _KNOWN_PRIMITIVE_TYPES[x]
+  elif sys.version_info >= (3, 9) and isinstance(x, types.GenericAlias):
+    # TODO(https://github.com/apache/beam/issues/23366): handle PEP 585
+    # generic type hints properly
+    raise TypeError(
+        'PEP 585 generic type hints like %s are not yet supported, '
+        'use typing module containers instead.' % x)
```

Comment:
Could we drop a link to either docs on typing module containers?

Reply:
Done